paper_framework_sources/p2079_system_execution_context.md

```diff
@@ -289,6 +300,48 @@ The authors do not have a recommendation, only a wish to see customisation avail
 We should decide how best to achieve it within the standard.
 Assuming we delegate customisation to the platform implementor, what wording would be appropriate for the specification, if any?
 
+
+## Extensibility
+
+The `std::execution` framework is expected to grow over time.
+We may need to add time-based scheduling, async I/O, priority-based scheduling, etc.
+The `system_context` framework needs to be designed in such a way that it allows for extensibility.
+
+Whatever the replaceability mechanism is, we need to ensure that new features can be added to the system context in a backwards-compatible manner.
+
+As the authors recommend an implementation-defined approach, the exact mechanism to allow for extensibility cannot be detailed here.
+
+
+## Shareability
+
+One of the motivations of this paper is to stop the proliferation of local thread pools, which can lead to thread oversubscription.
+If multiple binaries are used in the same process, we don't want each binary to have its own implementation of system context.
+Instead, we would want to share the same underlying implementation.
+
+The recommendation of this paper is to leave the details of shareability to be implementation-defined.
+
+## Performance
+
+As discussed above, one possible approach to the system context is to implement link-time replaceability.
+This implies moving across binary boundaries, over some defined API (which should be extensible).
+A common approach for this is to have COM-like objects.
+However, the problem with that approach is that it requires memory allocation, and memory allocation is a costly operation.
+This becomes problematic if we aim to encourage programmers to use the system context for spawning work in a concurrent system.
+
+While there are some costs associated with implementing all the goals stated here, we want the implementation of the system context to be as efficient as possible.
+For example, a good implementation should avoid memory allocation for the common case in which the default implementation is utilized for a platform.
+
+This paper cannot recommend the specific implementation techniques that should be used to maximize performance; these are considered Quality of Implementation (QOI) details.
+
+
+## Lifetime
+
+Underneath the `system_context`, there is a singleton of some sort. We need to specify the lifetime of this object and everything that derives from it.
+
+It makes sense to mandate that the lifetime of `system_context` underlying object should encompass the lifetime of `main()`.
+Using the `system_context` objects after the end of `main()` constitutes undefined behavior.
```

Comment:
Should it be constructed only when main starts too? I believe that discussions suggests that at some point. The reasoning being that if you launch async work there is a risk of it depending on as-yet-unconstructed objects. IF the expectation is that nothing will run until the start of main, users will get consistent behaviour WRT construction order.

What happens at the end of main? Do we define destruction/blocking behaviour at that point?